include/zephyr/drivers/display.h

```diff
@@ -367,11 +397,21 @@ static inline int display_blanking_off(const struct device *dev)
  * @retval 0 on success else negative errno code.
  * @retval -ENOSYS if not implemented.
  */
-static inline int display_set_brightness(const struct device *dev,
-					 uint8_t brightness)
+static inline int display_set_brightness(const struct device *dev, uint8_t brightness)
 {
-	struct display_driver_api *api =
-		(struct display_driver_api *)dev->api;
+	struct display_driver_api *api = (struct display_driver_api *)dev->api;
+	struct display_common_config *cfg = (struct display_common_config *)dev->config;
+
+#ifdef CONFIG_PWM
```

Comment:
Unless I'm missing something, this approach would require every display driver to define a gpio and PWM DT spec, even if that is not how the display sets its brightness.

Perhaps instead of this approach, we provide a common `display_set_brightness_gpio` function that can serve as an implementation of the `set_brightness` API? This `display_set_brightness_gpio` function would require that the caller have a `gpio_dt_spec` as the first entry in its config structure. This way, we don't require that all drivers have these structures but still benefit from code reuse.